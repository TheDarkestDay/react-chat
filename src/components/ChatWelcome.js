import React from 'react';
import PropTypes from 'prop-types';
import Paper from 'material-ui/Paper';
import Typography from 'material-ui/Typography';
import { withStyles } from 'material-ui/styles';

const styles = () => ({
  bold: {
    fontWeight: 'bold',
  },
  paper: {
    padding: '24px',
  },
  wrapper: {
    display: 'flex',
    flexGrow: 1,
    justifyContent: 'center',
    alignItems: 'center',
  },
});

const ChatWelcome = ({ classes }) => (
  <div className={classes.wrapper}>
    <Paper className={classes.paper}>
      <Typography variant="display1" color="textSecondary" gutterBottom>
        Start messaging...
      </Typography>
      <Typography variant="body1" color="inherit" gutterBottom>
        Use <span className={classes.bold}>Global</span> to explore communities around here.
      </Typography>
      <Typography variant="body1" color="inherit" gutterBottom>
        Use <span className={classes.bold}>Recents</span> to see your recent conversations.
      </Typography>
    </Paper>
  </div>
);

ChatWelcome.propTypes = {
  classes: PropTypes.objectOf(PropTypes.string).isRequired,
};

export default withStyles(styles)(ChatWelcome);
