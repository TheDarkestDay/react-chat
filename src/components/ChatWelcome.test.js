/* eslint-env jest */
import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import ChatWelcome from './ChatWelcome';

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<ChatWelcome />, div);
  ReactDOM.unmountComponentAtNode(div);
});

it('should match snapshot', () => {
  const tree = renderer.create(<ChatWelcome />).toJSON();
  expect(tree).toMatchSnapshot();
});
