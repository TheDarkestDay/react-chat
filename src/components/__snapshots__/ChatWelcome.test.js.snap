// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot 1`] = `
<div
  className="ChatWelcome-wrapper-57"
>
  <div
    className="MuiPaper-root-58 MuiPaper-shadow2-62 MuiPaper-rounded-59 ChatWelcome-paper-56"
  >
    <h1
      className="MuiTypography-root-85 MuiTypography-display1-89 MuiTypography-colorTextSecondary-107 MuiTypography-gutterBottom-102"
    >
      Start messaging...
    </h1>
    <p
      className="MuiTypography-root-85 MuiTypography-body1-94 MuiTypography-colorInherit-104 MuiTypography-gutterBottom-102"
    >
      Use 
      <span
        className="ChatWelcome-bold-55"
      >
        Global
      </span>
       to explore communities around here.
    </p>
    <p
      className="MuiTypography-root-85 MuiTypography-body1-94 MuiTypography-colorInherit-104 MuiTypography-gutterBottom-102"
    >
      Use 
      <span
        className="ChatWelcome-bold-55"
      >
        Recents
      </span>
       to see your recent conversations.
    </p>
  </div>
</div>
`;
